### DISCUSSION TITLE
Control Apple TV

### DISCUSSION THREAD

**Discussion by ghost** [nicko88/HTWebRemote#3] [Q&A] [answered]
Hey,
is there any way to control an Apple TV with this program. It would be the only thing missing in my setup.

**Comment by nicko88** [comment, marked as answer]
Sorry for not being able to provide direct support for ATV in HTWR.  The control method for an ATV is proprietary and encrypted and very complicated and others have done the work to reverse engineer it and keep it up to date with Apples software updates.

I would use [atvremote](https://pyatv.dev/documentation/atvremote/) option from [pyatv](https://pyatv.dev/).

If using Windows then create a .bat file such as this, to call the atvremote python app:

```
@echo off
atvremote -s 192.168.1.100 --companion-credentials {credential_string_here} %1
```

And then in HTWR for `pc` device, you can put in the path to the .bat file in the `cmd`, and the command to pass to `atvremote` in the `param`.

If using Linux, then do similar with a .sh script.

I have a friend using HTWR who controls his ATV this way and says it works well.


Or, since ATV includes an IR reciever you could get or make an IR blaster.

Either make a [Tasmota IR blaster](https://tasmota.github.io/docs/IR-Remote/) and use MQTT to control it.

Or get a network blaster like [Global Caché IP2IR iTach](https://www.amazon.com/Global-Cache-IP2IR-iTach-Wired/dp/B003BFTKUC) and use generic TCP device to control it.

Or get a [WinLIRC](https://winlirc.sourceforge.net/) compatible blaster like [IguanaIR](https://www.iguanaworks.net/products/usb-ir-transceiver/) and use the LIRC device to control it.

**Comment by ghost** [comment]
Thank you for your answer. I will look into these options.
